frontend/src/metabase/common/components/EntityPicker/components/ItemList/RootItemList.tsx

```diff
@@ -0,0 +1,123 @@
+import { useAsync } from "react-use";
+import { t } from "ttag";
+
+import { useCollectionQuery } from "metabase/common/hooks";
+import { PERSONAL_COLLECTIONS } from "metabase/entities/collections";
+import { useSelector } from "metabase/lib/redux";
+import { getUser, getUserIsAdmin } from "metabase/selectors/user";
+import type { CollectionId } from "metabase-types/api";
+
+import type {
+  EntityPickerOptions,
+  CollectionPickerItem,
+  TypeWithModel,
+  TisFolder,
+} from "../../types";
+
+import { ItemList } from "./ItemList";
+
+const personalCollectionsRoot: CollectionPickerItem = {
+  ...PERSONAL_COLLECTIONS,
+  can_write: false,
+  model: "collection",
+  location: "/",
+  description: "",
+};
+
+interface RootItemListProps<CollectionPickerItem extends TypeWithModel> {
+  onClick: (val: CollectionPickerItem) => void;
+  selectedItem: CollectionPickerItem | null;
+  itemName: string;
+  options: EntityPickerOptions;
+  isFolder: TisFolder<CollectionPickerItem>;
+  isCurrentLevel: boolean;
+}
+/**
+ * This is a special item list that exists "above" our analytics and might include:
+ * a) the highest-level collections the user can access (often "our analytics")
+ * b) the user's personal collection
+ * c) a top level folder including all personal collections (admin only)
+ */
+export const RootItemList = ({
+  onClick,
+  selectedItem,
+  itemName,
+  options,
+  isFolder,
+  isCurrentLevel,
+}: RootItemListProps<CollectionPickerItem>) => {
+  const isAdmin = useSelector(getUserIsAdmin);
+  const currentUser = useSelector(getUser);
+
+  const { data: personalCollection, isLoading: isLoadingPersonalCollecton } =
+    useCollectionQuery({
+      id: currentUser?.personal_collection_id,
+      enabled: !!currentUser?.personal_collection_id,
+    });
+
+  const {
+    data: rootCollection,
+    isLoading: isLoadingRootCollecton,
+    error: rootCollectionError,
+  } = useCollectionQuery({ id: "root" });
+
+  const { value: data, loading: isLoading } = useAsync(async () => {
```

Comment:
i don't think there's any async work happening here

Reply:
you're right! (there used to be, but we moved it out)